components/locid/src/helpers.rs

```diff
@@ -44,12 +45,88 @@ impl<T> ShortVec<T> {
         }
     }
 
+    #[inline]
+    pub fn as_mut_slice(&mut self) -> &mut [T] {
+        match self {
+            ShortVec::Empty => &mut [],
+            ShortVec::Single(v) => core::slice::from_mut(v),
+            ShortVec::Multi(v) => v.as_mut_slice(),
+        }
+    }
+
+    #[inline]
     pub const fn single(&self) -> Option<&T> {
         match self {
             ShortVec::Single(v) => Some(v),
             _ => None,
         }
     }
+
+    #[inline]
+    pub fn len(&self) -> usize {
+        match self {
+            ShortVec::Empty => 0,
+            ShortVec::Single(_) => 1,
+            ShortVec::Multi(ref v) => v.len(),
+        }
+    }
+
+    #[inline]
```

Comment:
Suggestion: since this function is not trivial, don't use `#[inline]`. The compiler will still inline it if it thinks it should, but it doesn't give an unnecessary hint otherwise.